### PR TITLE
feat(backend):Create crud organisation routes (AIILG-545)

### PR DESCRIPTION
**Pull request by tlawal86**
# Overview
Create crud routes within organisations ensuring they are solely accessible by system admins

## JIRA Ticket
https://mhclgdigital.atlassian.net/browse/AIILG-545

## Changes

- created CRUD routes for organisation entity alongside complimentary types
- test suite to validate the routes

## Testing (if applicable)

- Unit Tests

### CODE REVIEW

File: tests/backend/api/routes/test_templates.py [hunk 1]
```diff
@@ -19,7 +20,7 @@
 
 
 @pytest.mark.asyncio
-async def test_get_templates_success():
+async def test_get_templates_success(override_user, override_session):
     async with get_test_client() as ac:
```
Comment (tlawal86) [inline tests/backend/api/routes/test_templates.py:24]: This specific test would fail during ci runs. My reading is that previously userdep's (lack of) cleanup was inconsequential whereas now a lot more route based tests exist potentially causing conflicts. Added both params to prevent persistent test failures

File: common/database/postgres_models.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ class Organisation(BaseTableMixin, table=True):
     __tablename__ = "organisation"
     created_datetime: datetime = Field(sa_column=created_datetime_column(), default=None)
     updated_datetime: datetime = Field(sa_column=updated_datetime_column(), default=None)
-    name: str
+    name: str = Field(unique=True)
```
Comment (imad-chowdhury-dluhc) [inline common/database/postgres_models.py:121]: I think this'll need an associated alembic migration. Also some [context](https://github.com/communitiesuk/localai-local-transcribe/pull/183#discussion_r3194741078) on whether we want to make this unique or not.

File: tests/backend/api/routes/test_organisations.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    )
+
+    assert response.status_code == 204
+
+    mock_session.delete.assert_awaited_once_with(org)
+    mock_session.commit.assert_awaited_once()
+
+
+@pytest.mark.asyncio
+async def test_delete_organisation_not_found(
```
Comment (marianasoftwire) [inline tests/backend/api/routes/test_organisations.py:142]: Delete has its not-found case here, but the PATCH route's matching 404 branch isn't covered, worth adding a test_update_organisations_not_found mirror?

File: common/database/postgres_models.py (hunk 2 above)
Comment (tlawal86) [reply common/database/postgres_models.py:121]: Raised this during stand up with the conclusion being that the names should be unique. See [alembic file](https://github.com/communitiesuk/localai-local-transcribe/pull/206/changes#r3267250427) for related commentary.

File: alembic/versions/098a3bde2f4d_add_unique_constraint_to_organisations_.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+down_revision: Union[str, None] = 'fa95ee66e612'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_unique_constraint(None, 'organisation', ['name'])
+    # ### end Alembic commands ###
+
```
Comment (tlawal86) [inline alembic/versions/098a3bde2f4d_add_unique_constraint_to_organisations_.py:27]: Highlighting a point raised in discussion with Imad. If there are pre-existing entities with duplicated names, this migration script will fail. In practice, there aren't any orgs that exist yet but nevertheless worth highlighting.

File: tests/backend/api/routes/test_organisations.py (hunk 3 above)
Comment (tlawal86) [reply tests/backend/api/routes/test_organisations.py:142]: Agreed, added an additional test covering the 404 edge case

**Review by marianasoftwire — APPROVED** [review]
LGTM
